### PR TITLE
Feature/fleet mode rebase

### PR DESCRIPTION
**Pull request by garyschulte**
## PR description
rebase onto main of the services refactor for fleet mode, see original #6003

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/DefaultBlockchain.java [hunk 1]
```diff
@@ -393,6 +393,7 @@ private void cacheBlockData(final Block block, final List<TransactionReceipt> re
         cache -> cache.put(block.getHash(), block.getHeader().getDifficulty()));
   }
 
+  @SuppressWarnings("unused")
```
Comment (matkt) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/DefaultBlockchain.java:396]: I think we can remove it

**Review by matkt — COMMENTED** [review]
did we tried follower+leader with this last commit ?

**Comment by matkt** [comment]
@jframe could you check the block pruning modification we are doing in this PR to be sure it's correct. normally we did tests but I prefer to have your feedback

**Comment by garyschulte** [comment]
> @jframe could you check the block pruning modification we are doing in this PR to be sure it's correct. normally we did tests but I prefer to have your feedback

I inadvertentlty tested this feature on the captain of a captain-follower sepolia pair, and block pruning is definitely working.  

> did we tried follower+leader with this last commit ?
I am resyncing the follower now to ensure it works correctly, since it failed to find the pruned captain blocks on the prior test 😂

**Comment by garyschulte** [comment]
> did we tried follower+leader with this last commit ?

Yes, confirmed the follower syncs and follows with only a leader.  

Regarding the pruner on the follower, I see it configured, but I am not seeing any executions of the pruner triggered.

**Comment by jframe** [comment]
> @jframe could you check the block pruning modification we are doing in this PR to be sure it's correct. normally we did tests but I prefer to have your feedback

Is the changes in the ChainDataPruner to only start pruning after the initial sync? Think it's better not to delay this until after the initial sync to keep the storage used small. Ideally, you would have Besu configured with `--Xcheckpoint-post-merge-enabled` so there isn't anything to prune anyway. Probably need to think more about how these and sync all interact.

File: besu/VERSION_METADATA.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"besuVersion":"24.2.0-dev-c990209a"}
```
Comment (garyschulte) [inline besu/VERSION_METADATA.json:1]: need to add this file to .gitignore

**Review by matkt — APPROVED** [review]
if we had time to do a quick test with the fleet plugin I think we can merge this PR

**Comment by non-fungible-nelson** [comment]
🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢🚢

**Pull request merged**